conans/test/integration/command/install/install_local_package_test.py

```diff
@@ -0,0 +1,65 @@
+import os
+import platform
+
+from conans import load
+from conans.test.assets.genconanfile import GenConanfile
+from conans.test.utils.tools import TestClient
+
+
+def test_local_contents_and_generators():
+
+    client = TestClient()
+    hello = str(GenConanfile().with_name("hello").with_version("1.0")
+                .with_import("from conan.tools.files import save").with_import("import os"))
+    hello += """
+
+    def package(self):
+        save(self, os.path.join(self.package_folder, "my_hello.lib"), "contents")
+
+    def post_install(self):
+        save(self, os.path.join(self.package_folder, "my_hello_bis.lib"), "contents")
+    """
+    client.save({"conanfile.py": hello})
+    client.run("create .")
+
+    chat = str(GenConanfile().with_name("chat").with_version("1.0")
+               .with_import("from conan.tools.files import save").with_import("import os")
+               .with_requires("hello/1.0"))
+    chat += """
+    def package(self):
+        save(self, os.path.join(self.package_folder, "my_chat.lib"), "contents")
+    """
+    client.save({"conanfile.py": chat})
+    client.run("create .")
+
+    consumer = str(GenConanfile()
+                   .with_import("import os")
+                   .with_import("from conan.tools.files import save")
+                   .with_import("from conan.tools.layout import cmake_layout")
+                   .with_name("consumer")
+                   .with_version("1.0").with_requires("chat/1.0")
+                   .with_settings("os", "arch", "compiler", "build_type")
+                   .with_requirement("chat/1.0")
+                   .with_generator("CMakeDeps")
+               )
+    consumer += """
+    def layout(self):
+        cmake_layout(self)
+    """
+    client.save({"conanfile.py": consumer})
+    client.run("install . --local-folder -if=my_install")
```

Comment:
``my_install`` folder actually used for something?

Reply:
not really. Just hate to pollute the current dir even in tests (toc)

Reply:
I will remove for clarity

Reply:
Note: In case the layout doesn't declare generators folder it will use it.